core/src/main/java/org/elasticsearch/ingest/core/Processor.java

```diff
@@ -49,7 +54,6 @@
          * Implementations are responsible for removing the used keys, so that after creating a pipeline ingest can
          * verify if all configurations settings have been used.
          */
-        P create(Map<String, Object> config) throws Exception;
-
+        P create(String tag, Map<String, Object> config) throws Exception;
```

Comment:
downside is that each processor impl needs to fetch the tag, but maybe we can do this in a base processor class?